script.sublissimo/resources/lib/script.py

```diff
@@ -0,0 +1,639 @@
+        xbmcgui.Dialog().ok(_(32014), _(32056))
+        stretch_subtitle_menu(subtitlefile, filename)
+
+def make_timelines_classical(decimal):
+    hours = int(decimal / 3600000)
+    restminutes = decimal % 3600000
+    minutes = int(restminutes / 60000)
+    restseconds = restminutes % 60000
+    seconds = int(restseconds / 1000)
+    milliseconds = int(restseconds % 1000)
+    output = (str(hours).zfill(2) + ":" + str(minutes).zfill(2) + ":" +
+              str(seconds).zfill(2) + "," + str(milliseconds).zfill(3))
+    return output
+
+def sync_after_wizard(starting_time, ending_time, subtitlefile, filename):
+    start = make_timelines_classical(starting_time*1000)
+    end = make_timelines_classical(ending_time*1000)
+    current_start_sub = Subtitle(subtitlefile)
+    subtitlefile2 = current_start_sub.move_subtitles(start)
+    current_sub = Subtitle(subtitlefile2)
+    subtitlefile3 = current_sub.create_new_factor(end)
+    if subtitlefile3:
+        # Succes, Your subs starts at, your subs end at.
+        xbmcgui.Dialog().ok(_(32017), _(32036) + start + "\n" +
+                                      _(32037) + end)
+    show_dialog(subtitlefile3, filename)
+
+def retrieve_video(subtitlefile, filename):
+    global videodbfilename
+    if videodbfilename == None:
+        choice = xbmcgui.Dialog().contextmenu([_(32093), _(32094), _(32095), _(32005)])
+        if choice == 3 or choice == -1:
+            show_dialog(subtitlefile, filename)
+        pos_locations = ["videodb://movies/titles/", "videodb://tvshows/titles/", "videodb://"]
+        location = xbmcgui.Dialog().browse(1, _(32020), 'video', '', False, False, pos_locations[choice])
+        if location in pos_locations:
+            show_dialog(subtitlefile, filename)
+        videodbfilename = location    
+    else:
+        choice = xbmcgui.Dialog().contextmenu([_(32116) , _(32093), _(32094), _(32095), _(32005)])
+        if choice == 4 or choice == -1:
+            show_dialog(subtitlefile, filename)
+        pos_locations = [videodbfilename, "videodb://movies/titles/", "videodb://tvshows/titles/", "videodb://"]
+        if choice == 0:
+            location = videodbfilename
+        else:
+            location = xbmcgui.Dialog().browse(1, _(32020), 'video', '', False, False, pos_locations[choice])
+        if location in pos_locations[1:]:
+            show_dialog(subtitlefile, filename)
+        videodbfilename = location
+    return location
+
+def sync_with_video(subtitlefile, filename):
+    #Name, long desc, Ok, More Info
+    resp = xbmcgui.Dialog().yesno(_(31001), _(32060),
+                                   yeslabel=_(32012), nolabel=_(32013))
+    if not resp:
+        # How to, long desc.
+        xbmcgui.Dialog().textviewer(_(32061), _(32062))
+    location = retrieve_video(subtitlefile, filename)
+    xbmcPlayer = SyncWizard()
+    xbmcPlayer.add(subtitlefile, filename)
+    xbmcPlayer.play(location)
+    xbmc.sleep(500)
+    while xbmcPlayer.isPlaying():
```

Comment:
better to improve this at some point using monitor.waitForAbort. Although the pauses are small, monitor is designed to handle those cases